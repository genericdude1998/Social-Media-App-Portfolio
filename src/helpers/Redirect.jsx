import React from 'react';
import { Navigate } from 'react-router';
import PropTypes from 'prop-types';

export const Redirect = ({token, path}) => {
    return (token ? <Navigate to={path}/>:null);
}

Redirect.propTypes = {
    token: PropTypes.string,
    path:PropTypes.string.isRequired,
}