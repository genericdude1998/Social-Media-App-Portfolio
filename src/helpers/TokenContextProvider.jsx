import React, { createContext } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { mapTokenStateToProps } from '../view/mappers/mappers';

export const TokenContext = createContext();

const TokenProvider = ({token, children}) => 
    (
        <TokenContext.Provider value={'Sample Token'}>
            {children}
        </TokenContext.Provider>
    );

TokenProvider.propTypes = {
    token: PropTypes.string,
}

const ConnectedTokenProvider = connect(mapTokenStateToProps)(TokenProvider);

export default ConnectedTokenProvider;