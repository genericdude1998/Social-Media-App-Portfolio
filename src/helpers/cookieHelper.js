export function setCookie(name,value){
    const cookie =`${name}=${value}`;
    document.cookie = cookie;
    return cookie;
}