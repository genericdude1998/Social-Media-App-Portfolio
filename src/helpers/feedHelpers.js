export function refreshPosts(dispatch, doGetPostsThunk){
    dispatch(doGetPostsThunk());
}