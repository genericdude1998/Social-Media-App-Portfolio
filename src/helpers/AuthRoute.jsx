import React from 'react';
import { Route } from 'react-router-dom';
import { Navigate, Routes } from 'react-router';
import { TokenContext } from './TokenContextProvider';


const AuthRoute = ({element: Element}) => {
    const token = React.useContext(TokenContext);
    console.log(token);
    return (token ? Element : <Navigate to={'/notAuthorised'}/>
    );
}

export default AuthRoute;
