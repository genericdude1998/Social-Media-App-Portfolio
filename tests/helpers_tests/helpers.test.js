import { shallow } from 'enzyme';
import React from 'react';
import { Redirect } from '../../src/helpers/Redirect';
import { mockPath, mockToken } from '../mockValues';


describe('Redirect', () => {
    it('should matchSnapshot when not given token', () => {
        let wrapper = shallow(<Redirect token={undefined} path={mockPath}/>)
        expect(wrapper).toMatchSnapshot();
    });
    it('should matchSnapshot when given token', () => {
        let wrapper = shallow(<Redirect token={mockToken} path={mockPath}/>)
        expect(wrapper).toMatchSnapshot();
    });
});