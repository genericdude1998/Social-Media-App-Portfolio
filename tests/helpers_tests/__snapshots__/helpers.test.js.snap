// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirect should matchSnapshot when given token 1`] = `
<Navigate
  to="/"
/>
`;

exports[`Redirect should matchSnapshot when not given token 1`] = `""`;
